blackduck/Client.py

```diff
@@ -201,28 +198,56 @@ def get_metadata(self, name, parent=None, **kwargs):
         return self.get_resource(name, parent, items=False, **kwargs)
 
     def checked_request(self, method, url, name='', **kwargs):
-        """Request from endpoint and return json result
+        """Send request to url endpoint and return json result
 
         Args:
             method (str): http request type e.g. 'get', 'post', etc.
             url (str): of endpoint
-            name (str, optional): for informational purposes in case of error. Defaults to ''.
+            name (str, optional): for informational purposes in case of error. Defaults to None.
             kwargs: passed to session.request
 
         Returns:
-            json/dict: requested object
-        """
-        response = self.session.request(method, url, **kwargs)
-
-        if response.status_code != 200:
-            self.http_exception_handler(response, name)
+            json/dict: requested object (status code 200) else None
 
-        if 'Content-Type' in response.headers:
-            content_type = response.headers['Content-Type']
+        Raises:
+            requests.exceptions.HTTPError: from response.raise_for_status()
+        """
+        r = self.session.request(method, url, **kwargs)
+
+        status_code_lookup = {
+            200: "OK",
+            201: "Created",
+            204: "No Content",
+            401: "Unauthorized",
+            404: "Not Found",
+            406: "Not Acceptable",
+            412: "Precondition Failed",
+        }
+
+        if r.status_code not in [200, 201, 204]:
+            # print out a more descriptive error message before raising an exception
+            try:
+                content = json.dumps(r.json(), indent=4)
+            except json.JSONDecodeError:
+                logging.error("caught JSONDecodeError, using response.text")
+                content = r.text
+
+            logging.error(f"checked_request failed for name: {name}, url: {r.url}")
```

Comment:
I'm not a big fan of passing around argument 'name' in checked_request, get_items. Is it really required for debugging? It has no other purpose it appears.

Reply:
Previously used to identify object type before I wrote a reliable version of `blackduck.Utils.get_resource_name(obj)`